### PR TITLE
added image upload callbacks to imgmgr

### PR DESCRIPTION
**Pull request by JustineKH**
Added `start` `stop` and `complete` callbacks for imgmgr

### CODE REVIEW

File: mgmt/imgmgr/include/imgmgr/imgmgr.h [hunk 1]
```diff
@@ -111,7 +122,10 @@ int imgmgr_state_slot_in_use(int slot);
 int imgmgr_state_set_pending(int slot, int permanent);
 int imgmgr_state_confirm(void);
 int imgmgr_find_best_area_id(void);
-
+int imgmgr_register_callbacks(const imgmgr_dfu_callbacks_t *cb_struct);
+void imgr_dfu_stopped(void);
```
Comment (rymanluk) [inline mgmt/imgmgr/include/imgmgr/imgmgr.h]: was it intention to expose the `img_dfu_foo()`? If so it should have prefix `imgmgr`. If it was not the intention then those functions should be `static`

File: mgmt/imgmgr/src/imgmgr.c [hunk 2]
```diff
@@ -844,13 +853,47 @@ imgr_upload(struct mgmt_cbuf *cb)
     return imgr_upload_good_rsp(cb);
 }
 
+void
+imgr_dfu_stopped(void)
+{
+    if (imgmgr_dfu_callbacks_fn->dfu_stopped_cb != NULL)
```
Comment (rymanluk) [inline mgmt/imgmgr/src/imgmgr.c]: brackles here and below are not according to https://github.com/apache/mynewt-core/blob/master/CODING_STANDARDS.md

File: mgmt/imgmgr/src/imgmgr.c [hunk 3]
```diff
@@ -844,13 +853,47 @@ imgr_upload(struct mgmt_cbuf *cb)
...
+}
+
 void
 imgr_set_upload_cb(imgr_upload_fn *cb, void *arg)
 {
     imgr_upload_cb = cb;
     imgr_upload_arg = arg;
 }
 
+int
```
Comment (rymanluk) [inline mgmt/imgmgr/src/imgmgr.c]: maybe there is no need to return `int` here since we never fail.

**Comment by wes3** [comment]
Another comment: the pointer imgmgr_dfu_callbacks_fn should be checked for NULL before accessing. If someone does not call the register callbacks function the code will access a NULL pointer.

**Pull request merged**